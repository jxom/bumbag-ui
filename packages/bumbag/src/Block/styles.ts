import { cssClass } from '../styled';
import { theme } from '../utils';

export const Block = (styleProps) => cssClass`
  display: block;

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;
