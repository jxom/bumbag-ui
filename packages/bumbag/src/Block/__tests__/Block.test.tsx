import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Block } from '../Block';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Block>Hello world</Block>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(
      <Block position="absolute" top="0">
        Hello world
      </Block>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<Block use="p">Hello world</Block>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Block props', () => {
      const { result } = renderHook(() => Block.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(<Block>{(BlockProps) => <div {...BlockProps}>Hello world</div>}</Block>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('Block.base should render correctly', () => {
    const { container } = render(<Block>hello world</Block>, {
      theme: { Block: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly', () => {
    const { container } = render(<Block>hello world</Block>, {
      theme: { Block: { defaultProps: { className: 'test' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
