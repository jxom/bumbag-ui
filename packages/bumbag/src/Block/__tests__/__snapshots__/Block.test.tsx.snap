// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: block;
}

<p
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Block"
>
  Hello world
</p>
`;

exports[`composition hook should return with Block props 1`] = `
Object {
  "className": "css-oujebp css-63it0p css-jwhr8m bb-Box bb-Block",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: block;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Block"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: block;
}

<div
  class="emotion-0 emotion-1 emotion-2 test bb-Box bb-Block"
>
  hello world
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: block;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Block"
>
  Hello world
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: block;
}

.emotion-2 {
  position: absolute !important;
  top: 0 !important;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Block"
>
  Hello world
</div>
`;

exports[`theming Block.base should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  display: block;
}

.emotion-0 {
  background-color: var(--bb-palette-red,#ff0000);
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Block"
>
  hello world
</div>
`;
